plugins/Terminal/js/index.js

```diff
@@ -0,0 +1,224 @@
+import ReactDOM from 'react-dom'
+import { createStore } from 'redux'
+import { Provider } from 'react-redux'
+import rootReducer from './js/reducers/index.js'
+import CommandLine from './js/components/app.js'
+import Path from 'path'
+const fs = require('fs')
+const child_process = require('child_process')
+import { Map } from 'immutable'
+const http = require('http')
+const url = require('url')
+
+const checkSiaPath = () => new Promise((resolve, reject) => {
+	fs.stat(Path.resolve(SiaAPI.config.attr('siac').path, './siac'), (err) => {
+		 if (!err) {
+			resolve()
+		 } else {
+			reject(err)
+		 }
+	})
+})
+
+checkSiaPath().catch(() => {
+	// config.path doesn't exist.	Prompt the user for siad's location
+	if (!SiaAPI.config.attr('siac')) {
+		SiaAPI.config.attr('siac', { path: '' })
+	}
+	SiaAPI.showError({ title: 'Siac not found', content: 'Sia-UI couldn\'t locate siac. Please navigate to siac.' })
+	const siacPath = SiaAPI.openFile({
+		title: 'Please locate siac.',
+		properties: ['openFile'],
+		defaultPath: Path.join('..', SiaAPI.config.attr('siac').path || './' ),
+		filters: [{ name: 'siac', extensions: ['*'] }],
+	})
+	SiaAPI.config.attr('siac', { path: Path.dirname(siacPath[0]) })
+	SiaAPI.config.save()
+})
+
+const isCommandSpecial = function(commandString, specialArray) {
+	//Cleans string and sees if any subarray in array starts with the string when split.
+	var args = commandString.replace(/\s*\s/g, ' ').trim().split(' ')
+	if (args[0] === './siac' || args[0] === 'siac') {
+		args.shift()
+	}
+
+	//Can't do a simple match because commands can be passed additional arguments.
+	return specialArray.findIndex( (command) =>
+		command.reduce((matches, argument, i) =>
+			(matches && argument === args[i]),
+		true) && commandString.indexOf('-h') === -1 //Also covers --help.
+		//Don't show a password prompt if user is looking for help.
+	)
+}
+
+const spawnCommand = function(commandStr, actions) {
+	//Create new command object. Id doesn't need to be unique, just can't be the same for adjacent commands.
+
+	//We set the command first so the user sees exactly what they type. (Minus leading and trailing spaces, double spaces, etc.)
+	var commandString = commandStr.replace(/\s*\s/g, ' ').trim()
+	var newCommand = Map({ command: commandString, result: '', id: Math.floor(Math.random()*1000000), stat: 'running' })
+	actions.addCommand(newCommand)
+
+	//Remove surrounding whitespace and leading siac command.
+	if (commandString.startsWith('siac')) {
+		commandString = commandString.slice(4).trim()
+	} else if (commandString.startsWith('./siac')) {
+		commandString = commandString.slice(6).trim()
+	}
+
+	//Add address flag to siac.
+	var args = commandString.split(' ')
+	if (args.indexOf('-a') === -1 && args.indexOf('--address') === -1 && SiaAPI.config.attr('address')) {
+		args = args.concat([ '-a', SiaAPI.config.attr('address') ])
+	}
+
+	var siac = child_process.spawn('./siac', args, { cwd: SiaAPI.config.attr('siac').path })
```

Comment:
Users should rarely be using these commands from the terminal, so I'm not too worried about it. It would be a pain anyway, since the terminal plugin doesn't have filename autocomplete like a real terminal does.